redaction/draft-ietf-trans-redaction.md

```diff
@@ -0,0 +1,229 @@
+the subjectAltName extension is present, that the subjectAltName extension
+contains the same number of entries as the redactedSubjectAltName extension, and
+that each entry in the subjectAltName extension has a matching entry at the same
+position in the redactedSubjectAltName extension. Two entries are matching if
+either:
+
+* The two entries are identical; or
+* Both entries are DNS-IDs, have the same number of labels, and each label in
+  the subjectAltName entry has a matching label at the same position in the
+  redactedSubjectAltName entry. Two labels are matching if either:
+  * The two labels are identical; or,
+  * Neither label is `*` and the label from the redactedSubjectAltName entry is
+    equal to REDACT(label from subjectAltName entry) ({{redacting_labels}}).
+
+If any of these checks fail, the certificate MUST NOT be considered compliant.
+
+# Reconstructing the TBSCertificate    {#reconstructing_tbscertificate}
+
+Section TBD of [I-D.ietf-trans-rfc6962-bis] describes how TLS clients can
+reconstruct the TBSCertificate component of a precertificate from a certificate,
+so that associated SCTs may be verified.
+
+If the redactedSubjectAltName extension ({{redacted_san_extension}}) is present
+in the certificate, TLS clients MUST also:
+
+* Verify the redactedSubjectAltName extension against the subjectAltName
+  extension according to {{verifying_redacted_san}}.
+* Once verified, remove the subjectAltName extension from the TBSCertificate.
+
+# Security Considerations
+
+## Avoiding Overly Redacting Domain Name Labels
+
+Redaction of domain name labels carries the same risks as the use of wildcards
+(e.g., section 7.2 of [RFC6125]). If the entirety of the domain space below the
+unredacted part of a domain name is not registered by a single domain owner
+(e.g., REDACT(label).com, REDACT(label).co.uk and other [Public.Suffix.List]
+entries), then the domain name may be considered by clients to be overly
+redacted.
+
+CAs should take care to avoid overly redacting domain names in precertificates.
+It is expected that monitors will treat precertificates that contain overly
+redacted domain names as potentially misissued. TLS clients MAY consider a
+certificate to be non-compliant if the reconstructed TBSCertificate
+({{reconstructing_tbscertificate}}) contains any overly redacted domain names.
+      
+# Privacy Considerations    {#privacy_considerations}
+
+## Ensuring Effective Redaction
+
+Although the domain label redaction mechanism removes the need for private
+labels to appear in logs, it does not guarantee that this will never happen.
+Anyone who encounters a certificate could choose to submit it to one or more
+logs, thereby rendering the redaction futile.
+
+Domain owners are advised to take the following steps to minimize the likelihood
+that their private labels will become known outside their closed communities:
+
+* Avoid registering private labels in public DNS.
```

Comment:
Add:
* Avoid using publicly-trusted roots for securing a private DNS namespace
?